students/kristianjf/lesson04/mailroom_2.py

```diff
@@ -0,0 +1,167 @@
+the amounts they have donated. This structure should be populated at first with at \
+least five donors, with between 1 and 3 donations each.
+'''
+DATA_STRUCTURE = [('Jeff Bezos', [200, 500, 1000]), \
+                    ('Bill Gates', [1000]), \
+                    ('Steve Jobs', [20, 50, 100])]
+
+#PROGRAM_OPTIONS = ['Send a Thank You', 'Create a Report', 'Send letters to everyone', 'quit']
+
+def menu(options_dict):
+    '''
+    The script should prompt the user (you) to choose from a menu of 3 actions: \
+    “Send a Thank You”, “Create a Report” or “quit”)
+    '''
+    response = ''
+    options = [option[0:2] for option in enumerate(options_dict.keys(), 1)]
+    while True:
+        print('Please select a number from the list of the following options: \n')
+        for option in options:
+            print(option)
+        response = int(input())
+        response_selection = options[response-1][1]
+        options_dict[response_selection]()
+
+def thank_you():
+    '''
+    Sending a Thank You
+    If the user (you) selects ‘Send a Thank You’, prompt for a Full Name.
+        If the user types a name not in the list, add that name to the data structure and use it.
+        If the user types a name in the list, use it.
+    If the user types ‘list’, show them a list of the donor names and re-prompt
+    Once a name has been selected, prompt for a donation amount.
+    Turn the amount into a number – it is OK at this point for the program to crash if \
+    someone types a bogus amount.
+    Once an amount has been given, add that amount to the donation history of the selected user.
+    Finally, use string formatting to compose an email thanking the donor for their generous \
+    donation. Print the email to the terminal and return to the original prompt.
+    '''
+    program_options_ty = ['Send a Thank You', 'action: list', 'quit']
+    response = ''
+    options = [option for option in enumerate(program_options_ty, 1)]
+    while True:
+        print('Menu: Send a Thank You')
+        print('Please select a number or action from the list of the following options: \n')
+        for option in options:
+            print(option)
+        response = input()
+        if response == str(options[0][0]):
+            full_name = input('Please enter full name of the recipient: ').split()
+            '''
+            Format Full name
+            '''
+            full_name_cap = ''
+            for name in full_name:
+                full_name_cap += f'{name.capitalize()} '
+            full_name = full_name_cap.strip()
+            '''
+            Process Full Name
+            '''
+            ds_names = [name for name, donation in DATA_STRUCTURE]
+            if full_name in ds_names:
+                full_name_index = ds_names.index(full_name)
+                print(f'Found {full_name} in data_structure')
+                donation_amount = int(input('Please enter a donation amount: '))
```

Comment:
A minor bug, but casting this donation_amount to an int throws an error if someone's donated anything that isn't a specific dollar. I.e. if someone donates $12.56 for some reason.